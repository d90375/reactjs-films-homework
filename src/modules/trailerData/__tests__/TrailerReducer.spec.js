import TrailerReducer from "../TrailerReducer";

import {
  FETCH_TRAILER_FULFILLED,
  FETCH_TRAILER_PENDING,
  FETCH_TRAILER_REJECTED,
  FETCH_TRAILER_REMOVE
} from "../../actionTypes";

describe("modules > trailer > TrailerReducer", () => {
  it("returns initial state, if non matched action is dispatched", () => {
    const initialState = {
      trailer: {
        isLoadingTrailer: false,
        isFulfilledTrailer: false,
        hasErrorTrailer: false
      }
    };

    const action = {
      type: "FOO"
    };

    expect(TrailerReducer(initialState, action)).toBe(initialState);
  });

  it.each([[FETCH_TRAILER_FULFILLED], [FETCH_TRAILER_PENDING], [FETCH_TRAILER_REJECTED], [FETCH_TRAILER_REMOVE]])(
    `updates state according to dispatched action`,
    (actionType) => {
      const initialState = {
        value: 0
      };

      const payload =
        actionType === FETCH_TRAILER_FULFILLED
          ? {
              data: 1
            }
          : undefined;

      const action = {
        type: actionType,
        payload
      };

      expect(TrailerReducer(initialState, action)).toMatchSnapshot();
    }
  );
});
