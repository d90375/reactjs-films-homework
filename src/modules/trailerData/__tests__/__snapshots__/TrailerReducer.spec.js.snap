// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules > trailer > TrailerReducer updates state according to dispatched action 1`] = `
Object {
  "data": 1,
  "hasErrorTrailer": false,
  "isFulfilledTrailer": true,
  "isLoadingTrailer": false,
  "value": 0,
}
`;

exports[`modules > trailer > TrailerReducer updates state according to dispatched action 2`] = `
Object {
  "data": Object {},
  "hasErrorTrailer": false,
  "isFulfilledTrailer": false,
  "isLoadingTrailer": true,
  "value": 0,
}
`;

exports[`modules > trailer > TrailerReducer updates state according to dispatched action 3`] = `
Object {
  "errorTrailer": undefined,
  "hasErrorTrailer": true,
  "isFulfilledTrailer": false,
  "isLoadingTrailer": false,
  "value": 0,
}
`;

exports[`modules > trailer > TrailerReducer updates state according to dispatched action 4`] = `
Object {
  "data": Object {},
  "errorTrailer": Object {},
  "hasErrorTrailer": false,
  "isFulfilledTrailer": false,
  "isLoadingTrailer": false,
  "value": 0,
}
`;
